flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/FinishedOperatorChain.java

```diff
@@ -55,6 +55,11 @@ public boolean isFinishedOnRestore() {
         return true;
     }
 
+    @Override
+    public WatermarkGaugeExposingOutput<StreamRecord<OUT>> getMainOperatorOutput() {
+        return new FinishedOnRestoreMainOperatorOutput<>(streamOutputs);
+    }
+
```

Comment:
Maybe instead of introducing `FinishedOnRestoreMainOperatorOutput`, we could add `void OperatorChain#advanceToEndOfEventTime()` method?

Reply:
Hi Piotr~ I think semantically it would be better since it would avoid changing the meaning of `mainOperatorOutput`, but it currently seems to me that `MultipleInputStreamTask#advanceToEndOfEventTime()` has different implementation and we might not be able to have a unified implementation in the `OperatorChain`?

Reply:
Ok. I see, thanks for pointing this out. Let's keep it as it is. 

I think we need to follow up later on my previous suggestion, that instead of having `FinishedOperatorChain` that's used in different stream tasks, just have a single dedicated `FinishedStreamTask` that would be able to support any number of source/network inputs, without creating operator chain at all.